nixos/modules/services/x11/display-managers/default.nix

```diff
@@ -263,11 +268,24 @@ in
         '';
       };
 
-      extraSessionFilePackages = mkOption {
-        type = types.listOf types.package;
+      sessionPackages = mkOption {
+        type = with types; listOf (package // {
+          description = "package with provided sessions";
+          check = p: assertMsg
+            (package.check p && p ? passthru.providedSessions
+            && p.providedSessions != [] && all isString p.providedSessions)
+            ''
+              Package, '${p.name}', didn't specificy any session names, as strings, in
```

Comment:
Perhaps the message should only be asserted to the three last conjuncts – if `package.check p` fails, there might not be `p.name`.